File: 프로그래머스/LV2/코딩테스트 연습/연습문제/하노이의 탑/하노이의 탑.cpp
#include <string>
#include <vector>

using namespace std;

void go(int from, int to, int n, vector<vector<int>>& ans) {
    if(n == 0)return;

    int other = 6 - (from+to);
    go(from, other, n-1, ans);
    vector<int> temp = {from, to};
    ans.push_back(temp);
    go(other, to, n-1, ans);
}

vector<vector<int>> solution(int n) {
    vector<vector<int>> answer;
    go(1,3,n,answer);
    return answer;
}

File: 프로그래머스/LV2/코딩테스트 연습/연습문제/하노이의 탑/하노이의 탑.py
def go(frm, to, n, ans):
    if n == 0:
        return

    other = 6 - frm - to
    go(frm, other, n - 1, ans)
    ans.append([frm, to])
    go(other, to, n - 1, ans)
    
def solution(n):
    ans = []
    go(1, 3, n, ans)
    return ans